source/extensions/filters/http/jwt_authn/authenticator.cc

```diff
@@ -50,14 +62,18 @@ class AuthenticatorImpl : public Logger::Loggable<Logger::Id::filter>,
   // Calls the callback with status.
   void doneWithStatus(const Status& status);
 
-  // Return true if it is OK to forward this request without JWT.
-  bool okToBypass() const;
+  // Start verification process. It will continue to eliminate bad tokens until it finds one to
+  // verify with key.
+  void startVerify();
 
   // The config object.
-  FilterConfigSharedPtr config_;
+  const Protobuf::Map<ProtobufTypes::String, JwtProvider>& providers_;
```

Comment:
It seems that providers_ is for all providers in the config,  and it is only needed by sanitizePayloadHeaders().  

I think sanitize can be moved out of Authneticator class.  Ideally,  only sanitize headers for required providers.   Only sanitize all providers for allow_fail case where provider is not specified.

Reply:
moved to extractor.